services/cannon/src/Cannon/Run.hs

```diff
@@ -75,6 +79,27 @@ run o = do
       <*> mkClock
   refreshMetricsThread <- Async.async $ runCannon' e refreshMetrics
   s <- newSettings $ Server (o ^. cannon . host) (o ^. cannon . port) (applog e) m (Just idleTimeout)
+
+  -- Get the federaion domain list from Brig and start the updater loop
+  manager <- newManager defaultManagerSettings
+  let Brig bh bp = o ^. brig
+      baseUrl = BaseUrl Http (unpack bh) (fromIntegral bp) ""
+      clientEnv = ClientEnv manager baseUrl Nothing defaultMakeClientRequest
+      -- Loop the request until we get an answer. This is helpful during integration
+      -- tests where services are being brought up in parallel.
+      getInitialFedDomains = do
+        runClientM getFedRemotes clientEnv >>= \case
+          Right strat -> pure strat
+          Left err -> do
+            print $ "Could not retrieve the latest list of federation domains from Brig: " <> show err
+            threadDelay $ o ^. domainUpdateInterval
+            getInitialFedDomains
+  fedStrat <- getInitialFedDomains
+  tEnv <- newTVarIO fedStrat
+  let callback :: FederationDomainConfigs -> IO ()
+      callback = atomically . writeTVar tEnv
+  updateDomainsThread <- Async.async $ updateDomains clientEnv callback
```

Comment:
The TVar will need to be outside of the update loop so it can eventually be used by other code that I haven't updated yet. Otherwise it will be unreachable by the rest of the application.

Reply:
hm, might be, but this code is outdated anyway and wouldn't fit the tip of the branch any more.  i'll close this comment.